src/components/Table/utils/RowState.ts

```diff
@@ -10,11 +12,13 @@ import { applyRowFn, HEADER, KEPT_GROUP, matchesFilter, reservedRowKinds } from
  * We set up the RowStates in a tree, just like GridDataRow, to make business logic
  * that uses parent/children easier to write, i.e. selected-ness and collapsed-ness.
  */
-export class RowState {
-  /** Our row, only ref observed, so we don't crawl into GraphQL fragments. */
-  row: GridDataRow<any>;
+export class RowState<R extends Kinded> {
+  /** Our row, not actually observed, b/c each `createRows` calc creates unstable rows. */
+  private _row!: GridDataRow<R>;
+  /** Our data, only ref observed, so we don't crawl into GraphQL fragments. */
+  private _data: unknown;
```

Comment:
I split these up to better model our desired semantics of `rows` props, which are that:

* If you have a `createRows` that returns 100 new row literals, but
* All of them are just returning the same stable `data` fragment from the GQL cache

Then we won't treat that as a change, and won't re-render.

Prior to this PR, the `RowState.row` would actually change, which would mean things like `api.getSelectedRows` would think "I've not newly selected rows", b/c the row literals had changed, when really the selected _data_s had not changed.